authorization/role/service/role_management_service.go

```diff
@@ -38,3 +47,94 @@ func (r *RoleManagementServiceImpl) ListByResource(ctx context.Context, resource
 func (r *RoleManagementServiceImpl) ListAvailableRolesByResourceType(ctx context.Context, resourceType string) ([]role.RoleDescriptor, error) {
 	return r.repo.RoleRepository().FindRolesByResourceType(ctx, resourceType)
 }
+
+// assign assigns an identity ( users or organizations or teams or groups ) with a role, for a specific resource
+func (r *RoleManagementServiceImpl) assign(ctx context.Context, identityID uuid.UUID, resourceID string, roleRef rolerepo.Role) error {
+
+	ir := rolerepo.IdentityRole{
+		ResourceID: resourceID,
+		IdentityID: identityID,
+		RoleID:     roleRef.RoleID,
+	}
+
+	return r.repo.IdentityRoleRepository().Create(ctx, &ir)
+}
+
+// Assign assigns an identity ( users or organizations or teams or groups ) with a role, for a specific resource
+// IMPORTANT: This is a transactional method, which manages its own transaction/s internally
+func (r *RoleManagementServiceImpl) Assign(ctx context.Context, assignedBy uuid.UUID, identitiesToBeAssigned []uuid.UUID, resourceID string, roleName string) error {
+
+	rt, err := r.repo.ResourceRepository().Load(ctx, resourceID)
+
+	if err != nil {
+		return err
+	}
+
+	roleRef, err := r.repo.RoleRepository().Lookup(ctx, roleName, rt.ResourceType.Name)
+	if err != nil {
+		return err
+	}
+
+	// check if the current user token belongs to a user who has the necessary privileges
+	// for assigning roles to other users.
+
+	permissionService := permservice.NewPermissionService(r.repo)
+	hasScope, err := permissionService.HasScope(ctx, assignedBy, resourceID, authorization.ManageRoleAssignmentsInSpaceScope)
```

Comment:
I'm still a little uncomfortable about the scope name here.  Won't this method be used to assign roles for other types of resources, not just spaces?  E.g. assigning an admin role to a user for an organization.  I think what might be useful here is a helper method, perhaps in the `authorization` package, something like:

`func ManageRoleScopeForResourceType(resourceType string) string`

which accepts the name of a resource type as a parameter, and returns the name of the scope required to be able to manage roles for that resource type.

Reply:
Good point, how does this look https://github.com/fabric8-services/fabric8-auth/pull/481/commits/9d01ef67707f27d15566dfad39bc8b55eb009bf8 ?

Reply:
( hold on )

Reply:
How does this look :) https://github.com/fabric8-services/fabric8-auth/pull/481/files#diff-7dbcc3cc497b6deb6660e41e2eb25becR65